spyder/plugins/ipythonconsole/widgets/shell.py

```diff
@@ -237,51 +206,50 @@ def __init__(self, ipyclient, additional_options, interpreter_versions,
         # Show a message in our installers to explain users how to use
         # modules that don't come with them.
         self.show_modules_message = is_pynsist() or running_in_mac_app()
-        self.shutdown_lock = Lock()
 
     # ---- Public API ---------------------------------------------------------
-    def shutdown_kernel(self):
-        """Shutdown kernel."""
-        with self.shutdown_lock:
-            # Avoid calling shutdown_kernel on the same manager twice
-            # from different threads to avoid crash.
-            if self.kernel_manager.shutting_down:
-                return
-            self.kernel_manager.shutting_down = True
-        try:
-            self.kernel_manager.shutdown_kernel()
-        except Exception:
-            # kernel was externally killed
-            pass
+    @property
+    def is_spyder_kernel(self):
+        if self.kernel_handler is None:
+            return False
+        return self.kernel_handler.known_spyder_kernel
+
+    def connect_kernel(self, kernel):
+        """Connect to kernel."""
+        # Kernel client
+        kernel_client = kernel.kernel_client
+        kernel_client.stopped_channels.connect(self.notify_deleted)
+        kernel_client.start_channels()
+        self.kernel_client = kernel_client
+
+        self.kernel_manager = kernel.kernel_manager
+        self.kernel_handler = kernel
+        # Send message to kernel to check status
+        self.check_spyder_kernel()
+        self.sig_shellwidget_created.emit(self)
+
+    def notify_deleted(self):
+        """Notify that the shellwidget was deleted."""
+        self.sig_shellwidget_deleted.emit(self)
 
     def shutdown(self, shutdown_kernel=True):
         """Shutdown connection and kernel."""
         if self.shutting_down:
             return
         self.shutting_down = True
-        if shutdown_kernel:
-            if not self.kernel_manager:
-                return
+        self.close_kernel(shutdown_kernel)
+        super(ShellWidget, self).shutdown()
```

Comment:
```suggestion
        super().shutdown()
```

That was the old Python 2 syntax for `super`.